janus_messages/src/lib.rs

```diff
@@ -1104,13 +1109,397 @@ impl Decode for CollectResp {
+impl Encode for AggregateContinueResp {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        encode_u16_items(bytes, &(), &self.prepare_steps);
+    }
+}
+
+impl Decode for AggregateContinueResp {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let prepare_steps = decode_u16_items(&(), bytes)?;
+        Ok(Self { prepare_steps })
+    }
+}
+
+/// DAP protocol message representing a request from the leader to a helper to provide an
+/// encrypted aggregate of its share of data for a given batch interval.
+#[derive(Clone, Derivative, PartialEq, Eq)]
+#[derivative(Debug)]
+pub struct AggregateShareReq {
+    pub task_id: TaskId,
+    pub batch_interval: Interval,
+    #[derivative(Debug = "ignore")]
+    pub aggregation_param: Vec<u8>,
+    pub report_count: u64,
+    pub checksum: NonceChecksum,
+}
+
+impl AggregateShareReq {
+    /// The media type associated with this protocol message.
+    pub const MEDIA_TYPE: &'static str = "application/dap-aggregate-share-req";
+}
+
+impl Encode for AggregateShareReq {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        self.task_id.encode(bytes);
+        self.batch_interval.encode(bytes);
+        encode_u16_items(bytes, &(), &self.aggregation_param);
+        self.report_count.encode(bytes);
+        self.checksum.encode(bytes);
+    }
+}
+
+impl Decode for AggregateShareReq {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let task_id = TaskId::decode(bytes)?;
+        let batch_interval = Interval::decode(bytes)?;
+        let agg_param = decode_u16_items(&(), bytes)?;
+        let report_count = u64::decode(bytes)?;
+        let checksum = NonceChecksum::decode(bytes)?;
+
+        Ok(Self {
+            task_id,
+            batch_interval,
+            aggregation_param: agg_param,
+            report_count,
+            checksum,
+        })
+    }
+}
+
+/// DAP protocol message representing a helper's response to the leader's request to provide an
+/// encrypted aggregate of its share of data for a given batch interval.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct AggregateShareResp {
+    pub encrypted_aggregate_share: HpkeCiphertext,
+}
+
+impl AggregateShareResp {
+    /// The media type associated with this protocol message.
+    pub const MEDIA_TYPE: &'static str = "application/dap-aggregate-share-resp";
+}
+
+impl Encode for AggregateShareResp {
+    fn encode(&self, bytes: &mut Vec<u8>) {
+        self.encrypted_aggregate_share.encode(bytes);
+    }
+}
+
+impl Decode for AggregateShareResp {
+    fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
+        let encrypted_aggregate_share = HpkeCiphertext::decode(bytes)?;
+
+        Ok(Self {
+            encrypted_aggregate_share,
+        })
+    }
+}
+
+#[cfg(feature = "test-util")]
+pub mod test_util {
+    use super::{Nonce, Report, TaskId, Time};
+    use rand::{thread_rng, Rng};
+
+    pub fn new_dummy_report(task_id: TaskId, when: Time) -> Report {
```

Comment:
I think this method (and in general, all test-util functionality that is not used in `janus_messages` tests?) should live in Janus.